en/engine/event/event-builtin.md

```diff
@@ -0,0 +1,167 @@
+
+At the same time, if C has a parent node, it will also pass the touch event to its parent node through the event bubble mechanism.
+
+### Point of Contact Attribution for Different Canvas
+
+Contact interception between different Canvas is determined by priority. In the scene in the figure below, Canvas 1-5 in the node tree corresponds to priority 1-5 of the image display. It can be seen that even though Canvas nodes 3, 4, and 5 are arranged in scrambled order, the order of response of the contacts is still **Canvas5 -> Canvas4 -> Canvas3 -> Canvas2 -> Canvas1**, according to the priority relationship on the Canvas. The sorting between Canvas is done in the order of the node tree only if the priority is the same.
+
+![multi-canvas](multi-canvas.png)
+
+### Register touch or mouse events in the capturing phase
+
+Sometimes it is needed to dispatch the touch or mouse events to parent node event listeners before dispatching to any child nodes beneath it in hierarchy, like the design of __cc.ScrollView__ component.
+
+Now the event bubbling can't meet our demand, so that we need to register the parent node event listeners in the capturing phase. To achieve this goal, passing in `true` as the `useCapture` parameter (a fourth parameter) when registering touch or mouse event on the node. For example:
+
+```ts
+this.node.on(Node.EventType.TOUCH_START, this.onTouchStartCallback, this, true);
+```
+
+When the node fires the `touch-start` event, the `touch-start` event will be first dispatched to all the parent node event listeners registered in the capturing phase, then dispatched to the node itself, and finally comes the event bubbling phase.
+
+Only touch or mouse events can be registered in the capturing phase, while the other events can't be.
+
+### Event Interception
+
+Normal events are dispensed as described above. However, if the node has components such as __Button__, __Toggle__ or __BlockInputEvents__ on it, it will stop event bubbling. Look at the picture below. There are two buttons, priority 1 for Canvas0 and priority 2 for Canvas1. If you click on the intersection of the two buttons, which is the blue area in the picture, it appears that button priority 2 received the contact event successfully, while button priority 1 did not. That's because according to the event reception rules above, button priority 2 receives contact events first and intercepts them（`event.propagationStopped = true`）to prevent event penetration. If the node is a non-button node, events can also be intercepted by adding the __BlockInputEvents__ component to prevent penetration.
+
+![events-block](events-block.png)
+
+## Example for touch events
+
+Let's make a summary of touch event propagation with the example below. There are four nodes A, B, C and D in the picture above, where A and B are brother nodes.
```

Comment:
Let's make a summary of touch event propagation with the example below.
-->
Using the example below, summarizing touch events is easy.